clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3151,11 +3151,21 @@ static void handleWorkGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
       return;
     ZDimExpr = ZDim.get();
 
+    // If the declaration has an [[intel::num_simd_work_items()]] attribute,
+    // check to see if the first argument (OpenCL spelling) or last argument
+    // (SYCL spelling) can be evenly divided by the num_simd_work_items
+    // attribute since first and last argument are only swapped for the
+    // Intel SYCL attributes and not the OpenCL ones.
     if (const auto *A = D->getAttr<SYCLIntelNumSimdWorkItemsAttr>()) {
       int64_t NumSimdWorkItems =
           A->getValue()->getIntegerConstantExpr(Ctx)->getSExtValue();
 
-      if (XDimVal.getZExtValue() % NumSimdWorkItems != 0) {
+      unsigned checkWorkGroupSize =
```

Comment:
```suggestion
      unsigned CheckWorkGroupSize =
```
per usual coding conventions.

Reply:
Done